evap/staff/importers/user.py

```diff
@@ -0,0 +1,384 @@
+        for user in users_with_same_names:
+            msg += format_html("<br /> - {} ({})", self._create_user_string(user), _("existing"))
+        msg += format_html("<br /> - {} ({})", self._create_user_string(user_data), _("import"))
+
+        self.importer_log.add_warning(msg, category=ImporterLogEntry.Category.DUPL)
+
+    @staticmethod
+    def _create_user_string(user: Union[UserProfile, UserData]):
+        return format_html("{} {} {}, {}", user.title or "", user.first_name, user.last_name, user.email or "")
+
+
+class UserDataValidationChecker(Checker):
+    """Run django validation against UserData instances"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Only give one error per unique user_data.
+        self.already_checked: Set[UserData] = set()
+
+    def check_userdata(self, user_data: UserData, _location: ExcelFileLocation):
+        if user_data.email == "":
+            # Should trigger another checker. We cannot meaningfully give an error message.
+            return
+
+        if user_data in self.already_checked:
+            return
+        self.already_checked.add(user_data)
+
+        try:
+            user_data.validate()
+        except ValidationError as e:
+            self.importer_log.add_error(
+                _("User {user_email}: Error when validating: {error}").format(user_email=user_data.email, error=e),
+                category=ImporterLogEntry.Category.USER,
+            )
+
+
+class DuplicateUserDataChecker(Checker):
+    """Check for duplicate users"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.first_location_by_user_data: Dict[UserData, ExcelFileLocation] = {}
+
+        self.tracker: FirstLocationAndCountTracker = FirstLocationAndCountTracker()
+
+    def check_userdata(self, user_data: UserData, location: ExcelFileLocation):
+        if user_data not in self.first_location_by_user_data:
+            self.first_location_by_user_data[user_data] = location
+            return
+
+        first_location = self.first_location_by_user_data[user_data]
+        self.tracker.add_location_for_key(location=location, key=first_location)
+
+    def finalize(self):
+        for first_location, location_string in self.tracker.aggregated_keys_and_location_strings():
+            self.importer_log.add_warning(
+                _("{location}: The duplicated row was ignored. It was first found at {first_location}.").format(
+                    location=location_string,
+                    first_location=first_location,
+                ),
+                category=ImporterLogEntry.Category.IGNORED,
+            )
+
+
+# TODO: Not a real class that we want to instantiate --> make a set of functions? At least make functions private.
+class UserImporter:
+    class UserDataAdapter(RowCheckerMixin):
+        """Adapter to use UserDataChecker with UserParsedRow"""
```

Comment:
There is no one `UserDataChecker` (anymore?)

Reply:
changed wording to "Checker for UserData" -- or would you rather have the ABC `UserDataChecker` back? I removed it to reduce class count.

Reply:
Nope, now it's good :+1: